bonfire/bonfire.py

```diff
@@ -1000,6 +1001,8 @@ def _check_and_reserve_namespace(name, requester, duration, pool, timeout, local
     if not has_ns_operator():
         _error(f"{NO_RESERVATION_SYS}")
 
+    get_namespace_pools()
```

Comment:
Here you should check the result of this function call to make sure the 'pool' that the user is trying to use is valid.

For ex:

```python
    if pool not in get_namespace_pools():
        _error(f"namespace pool '{pool}' does not exist on this cluster")
```